doc/md/declarative/declarative-hcl.mdx

```diff
@@ -0,0 +1,225 @@
+---
+id: declarative-hcl-singleschema
+title: A Quickstart Guide to Declarative Migrations with a Single Schema
+slug: /declarative/declarative-hcl
+---
+
+import InstallationInstructions from '../components/_installation_instructions.mdx'
+
+### Introduction
+
+Atlas makes it easy to manage your database using the declarative migrations workflow. Declarative migrations focus on describing the desired state of the database and allow the migration engine to plan and execute the necessary actions to reach that state.
+
+In this quickstart guide, we will go through the three essential steps for using Atlas to manage your database schema: inspecting your database, authoring the desired changes, and applying those changes.
+
+### Installing Atlas
+
+<InstallationInstructions />
+
+### Warming up
+To demonstrate the examples in the article, let's create a docker container with an `example` database and insert a table in it.
+Use the following command to run a MySQL Docker container
+
+```console
+docker run --name atlas-demo -e MYSQL_ROOT_PASSWORD=pass -e MYSQL_DATABASE=example -p 3306:3306 -d mysql:latest
+```
+
+Once the container is running, you can connect to the MySQL server using the statement below:
+
+```console
+docker exec -it atlas-demo mysql -ppass
+```
+
+Once you're connected, you can switch to the `example` database and add a table in it by running the following command:
+
+```sql
+USE example;
+CREATE TABLE customers (
+    id INT PRIMARY KEY AUTO_INCREMENT,
+    first_name VARCHAR(255) NOT NULL,
+    last_name VARCHAR(255) NOT NULL
+);
+```
+
+## Step 1: Inspecting the database
+
+Before making any changes to your database, it's important to know what your database schema looks like. To do this, you can use the `atlas schema inspect` command to inspect your database. This command generates a file that represents your database schema, which can be used as a starting point for making changes.
```

Comment:
```suggestion
Before planning any changes to your database, it can be useful to know what your database schema looks like. To do this, you can use the `atlas schema inspect` command to inspect your database. This command generates a file that represents your database schema, which can be used as a starting point for making changes.
```

Reply:
also link to schema inspect full doc